components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -311,68 +312,115 @@ public void deleteUser(String userId) throws NotFoundException, CharonException
                 requiredAttributes);
     }
 
-    private List<Object> listUsers(Map<String, Boolean> requiredAttributes) throws CharonException {
+    private List<Object> listUsers(Map<String, Boolean> requiredAttributes, int offset, int limit, String sortBy,
+            String sortOrder, String domainName) throws CharonException {
 
-        ClaimMapping[] coreClaims;
-        ClaimMapping[] userClaims;
-        ClaimMapping[] extensionClaims = null;
         List<Object> users = new ArrayList<>();
-        //0th index is to store total number of results;
+        //0th index is to store total number of results.
         users.add(0);
+
+        String[] userNames;
+
+        if (StringUtils.isNotEmpty(domainName)) { // When domain is not null or not empty.
+            userNames = getAllUsernamesByDomainName(domainName);
+        } else { // When domain is null or empty.
+            userNames = getAllUsernamesByAllDomainNames();
+        }
+
+        if (ArrayUtils.isEmpty(userNames)) {
+            if (log.isDebugEnabled()) {
+                log.debug("There is no user who comply with the requested conditions.");
```

Comment:
shall we add context information to make the log more meaningful?

Reply:
It should be like... "No matching users found for condition $$$"

Reply:
Fixed